ttfi-core/src/main/scala/TTFI.scala

```diff
@@ -434,19 +485,21 @@ object TTFI {
               case _ => Left(s"Parse error in ${x}")
             }
           }
-          def fromTree[repr[+_]](x: Tree[Integer])(implicit s1: ExpSym[repr]): Either[ErrMsg, repr[Integer]] =
+          def fromTree[repr[_]](x: Tree[Integer])(implicit s1: ExpSym[repr]): Either[ErrMsg, repr[Integer]] =
             Fix(fromTree_[repr])((x, s1))
 
-          def fromTreeExt_[repr[+_]](self: ((Tree[Integer], MulSym[repr], ExpSym[repr])) => Either[ErrMsg, repr[Integer]])(args: (Tree[Integer], MulSym[repr], ExpSym[repr])): Either[ErrMsg, repr[Integer]] = args match {
-            case (x, s1, s2) => x match {
-              case Node("Mul", Seq(x, y)) => for {
-                a <- self((x, s1, s2)).right
-                b <- self((y, s1, s2)).right
-              } yield s1.mul(a)(b)
-              case x => fromTree_((x: (Tree[Integer], ExpSym[repr])) => self(x._1, s1, x._2))((x, s2))
-            }
-          }
-          def fromTreeExt[repr[+_]](x: Tree[Integer])(implicit s1: MulSym[repr], s2: ExpSym[repr]): Either[ErrMsg, repr[Integer]] = Fix(fromTreeExt_[repr])((x, s1, s2))
+          // def fromTreeExt_[repr[_]](self: ((Tree[Integer], MulSym[repr], ExpSym[repr])) => Either[ErrMsg, repr[Integer]]
+          //   )(args: (Tree[Integer], MulSym[repr], ExpSym[repr])): Either[ErrMsg, repr[Integer]] = args match {
+          //   case (x, s1, s2) => x match {
+          //     case Node("Mul", Seq(x, y)) => for {
+          //       a <- self((x, s1, s2)).right
+          //       b <- self((y, s1, s2)).right
+          //     } yield s1.mul(a)(b)
+
+          //     case x => fromTree_((x: (Tree[Integer], ExpSym[repr])) => self(x._1, s1, x._2))((x, s2))
```

Comment:
"-Xlint" gives
```
[warn] xxxxxxxxxxxxxxxxxxxxxxx/Projects/TTFI_su/ttfi-core/src/main/scala/TTFI.scala:497: Adapting argument list by creating a 3-tuple: this may not be what you want.
[warn]         signature: Function1.apply(v1: T1): R
[warn]   given arguments: x._1, s1, x._2
[warn]  after adaptation: Function1((x._1, s1, x._2): (TTFI.Final.TreeSem.Tree[Integer], TTFI.Final.MulSym.MulSym[repr], TTFI.Final.ExpSym.ExpSym[repr]))
[warn]               case x => fromTree_((x: (Tree[Integer], ExpSym[repr])) => self(x._1, s1, x._2))((x, s2))
[warn]                                                                             ^
[warn] one warning found
```